### ISSUE TITLE
android_alarm_manager fails to run in background with uninitialized AlarmService

### ISSUE THREAD

**Issue by deckerst** [flutter/flutter#21766]
Trying to run the android_alarm_manager v0.2.1 sample, the callbacks work fine when the app is still in foreground. However, if I launch the app then kill it, the callbacks are not executed in the background, because every time they should be, the AlarmService logs this instead:
`I/AlarmService: AlarmService has not yet started.`

It seems the AlarmService needs to be initialized with the following (by the way, this instruction is missing from the plugin README):
`await AndroidAlarmManager.initialize();`

Initializing the service in `main()` allows the service to work in the foreground case, but not in the background case.

My expectation is that the android_alarm_manager plugin allows execution of callbacks even when the app is no longer in the foreground. Is it incorrect?

Is there a way to have the service initialized when needed in the background?

flutter doctor -v
```
[✓] Flutter (Channel dev, v0.8.2, on Mac OS X 10.13.6 17G65, locale en-KR)
    • Flutter version 0.8.2 at /Users/thibaultdeckers/projects/flutter
    • Framework revision 5ab9e70727 (5 days ago), 2018-09-07 12:33:05 -0700
    • Engine revision 58a1894a1c
    • Dart version 2.1.0-dev.3.1.flutter-760a9690c2

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/thibaultdeckers/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.2

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 28.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] VS Code (version 1.27.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.18.0

[✓] Connected devices (2 available)
    • E5823     • CB5A28Q7EL                           • android-arm64 • Android 7.1.1 (API 25)
    • iPhone 6s • BBC95D90-23F8-4800-84D5-A61344682BD3 • ios           • iOS 11.4 (simulator)

• No issues found!
```

**Comment by deckerst** [comment]
I haven't found a solution at the time and I'm not working with Flutter these days. It seems Flutter is not ready yet for complex background scenarios, whether on iOS or Android. In the meantime, you could upvote this issue to give it more visibility.

**Comment by alexandranb** [comment]
I see, I think I'm gonna create my own plugin for alarm service with native Java code for Android, as I really need this to work, as soon as possible.
Anyway, thanks. It's good to see at least that are other people with this problem too, and it's not your fault for maybe not integrating correctly the plugin, but the plugin itself.

Thanks

**Comment by bkonyi** [comment]
Hi all,

Sorry for the delay on this issue. This should be fixed once flutter/plugins#1129 lands and `android_alarm_manager` 0.4.0 is published. On upgrading, the `AndroidManifest.xml` will need to be updated with the following:

  ```xml
        <service
            android:name="io.flutter.plugins.androidalarmmanager.AlarmService"
            android:permission="android.permission.BIND_JOB_SERVICE"
            android:exported="false"/>
        <receiver
            android:name="io.flutter.plugins.androidalarmmanager.AlarmBroadcastReceiver"
            android:exported="false"/>
  ```

**Comment by bkonyi** [comment]
This should be resolved by flutter/plugins#1129.
